torch/_inductor/codegen/cuda/cuda_cpp_scheduling.py

```diff
@@ -114,3 +148,93 @@ def codegen_template(
 
         V.graph.removed_buffers |= kernel.removed_buffers
         self.free_buffers_in_scheduler()
+
+    @staticmethod
+    def _unwrap_epilogue_nodes(
+        fused_node: FusedSchedulerNode,
+    ) -> list[BaseSchedulerNode]:
+        nodes = list(fused_node.get_nodes())
+        template_node = fused_node.get_template_node()
+        assert all(n.node is not None for n in nodes), (
+            "All epilogue nodes should have an IRNode"
+        )
+        return cast(
+            list[BaseSchedulerNode], [n for n in nodes if n.node is not template_node]
+        )
+
+    def _can_fuse_epilogue_impl(
+        self,
+        cuda_template_buffer: CUDATemplateBuffer,
+        epilogue_nodes: list[BaseSchedulerNode],
+        additional_node: BaseSchedulerNode,
+    ) -> bool:
+        """
+        Check if the given node can be fused with the epilogue. At the moment, Kernels
+        support fusion with Pointwise operations, wrapped in (named) ComputedBuffer nodes.
+
+        Args:
+            cuda_template_buffer : A CUDATemplateBuffer object representing the CUDA template and it's result buffer
+            epilogue_nodes : List[ir.Buffer]: The list of already fused epilogue nodes.
+            additional_node: The ir.Buffer node to be checked if it can be fused with the epilogue.
+        Returns:
+        - bool: True if the given node can be fused with the epilogue, False otherwise.
+
+        """
+        additional_ir_node = additional_node.node
+
+        if not isinstance(cuda_template_buffer, CUDATemplateBuffer):
+            return False
+        # if not cuda_template_buffer.template.can_fuse_epilogue:
+        #    # The used GEMM op does not support fusing epilogues
+        #    return False
+        if not isinstance(additional_ir_node, ComputedBuffer):
+            return False
+        if not isinstance(additional_ir_node.data, Pointwise):
+            return False
+        # We can fuse a Pointwise op that depends on the last fused epilogue node
+        # if any. If there is no epilogue node yet, it needs to depend on the template
+        # node
+        node_name = additional_ir_node.get_computed_buffer_name()  # type: ignore[attr-defined]
+        if node_name is None:
+            return False
+
+        if len(epilogue_nodes) == 0:
+            if cuda_template_buffer.name not in additional_ir_node.get_read_names():
+                return False
+        else:
+            last_epilogue_node = epilogue_nodes[-1].node
+            assert isinstance(last_epilogue_node, ComputedBuffer)  # for mypy
```

Comment:
Why do we have `epilogue_nodes: list[BaseSchedulerNode]` above then Comptued Buffer here ?

Reply:
Yeah I think this was previously ready for review, but I've butchered it since, sorry! Thanks for taking a look though. The comments are helpful regardless. Will fix.

Reply:
I removed this check, because it's possible the new node might depend on a node other than the last epilogue node.